src/components/marker-chart/Canvas.js

```diff
@@ -114,9 +118,27 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
           ? undefined
           : markerIndexToTimingRow.get(hoveredItem);
 
-      if (newRow !== undefined) {
+      if (newRow !== undefined && newRow !== rightClickedRow) {
+        /**
+         * To avoid multiple renders on the same row rectangle in canvas, first the row
+         * is cleared and then redrawn on using the drawMarkers()
+         **/
```

Comment:
I don't understand this comment very well :-)

Reply:
Initially, if we'd hover over markers that are in the same row the `fillStyle()` would fill on the row multiple times and consequently make the row darker. This comment was to explain why we `clearRect()` (which now is replaced by `clearRow()` )and then do `drawMarkers()`.